src/mds/journal.cc

```diff
@@ -237,6 +237,13 @@ void LogSegment::try_to_expire(MDS *mds, MDSGatherBuilder &gather_bld, int op_pr
     mds->sessionmap.save(gather_bld.new_sub(), sessionmapv);
   }
 
+  // updates to sessions for completed_requests
+  for (std::set<entity_name_t>::iterator i = touched_sessions.begin();
+       i != touched_sessions.end(); ++i) {
+    mds->sessionmap.save_if_dirty(*i, &gather_bld);
+  }
+  touched_sessions.clear();
+
```

Comment:
hmm, I don't want to add it into ::save because it would potentially make the rados ops from there larger than mds_sessionmap_keys_per_op.  But yeah, maybe we do need to do some batching here...